OMP Hobby ZMO integration documentation (#2772)
PX4/PX4-user_guide: en/frames_vtol/vtol_tiltrotor_omp_hobby_zmo_fpv.md

# OMP Hobby ZMO FPV

The OMP Hobby ZMO is a small sized tiltrotor VTOL that is available as RTF kit.

![Finished ZMO hover 1](../../assets/airframes/vtol/omp_hobby_zmo_fpv/airframe-hover.jpg)

![Finished ZMO hover 2](../../assets/airframes/vtol/omp_hobby_zmo_fpv/airframe-hover2.jpg)

## Overview

Key airframe features:

- Spacious interior
- Pre installed actuators
- Quick release wing connecting system
- ~35 minute long flight times (depending on takeoff weight)
- VTOL enables flying in locations where a fixed wing couldn't fly
- Battery and charger included in the kit
- Easy overall build
- Space to mount FPV and/or action camera in the front
- Airspeed data from pitot/static probe
- Lidar distance sensor

Depending on the final takeoff weight the hover time might be limited, since there is not a lot of air circulation inside the fuselage when the vehicle is hovering.

> ## Where to buy

Reviewer: We use "First Letter Capitalization". I'll modify

```suggestion
## Where to Buy

```

- [OMP-Hobby](https://www.omphobby.com/OMPHOBBY-ZMO-VTOL-FPV-Aircraft-With-DJI-Goggles-And-Remote-Controller-p3069854.html)
- [GetFPV](https://www.getfpv.com/omphobby-zmo-z3-vtol-fpv-1200mm-arf-plane-kit-no-fpv-system.html)
- [FoxtechFPV](https://www.foxtechfpv.com/zmo-pro-fpv-vtol.html)

> ## Flight controller options

Reviewer: Note, there are many many more options than this. SO let's be more clear and state that these are options that have been tried.

```suggestion
## Flight Controller

The following options have been tested:
```

- [Auterion Skynode evaluation kit](https://auterion.com/product/skynode/)
- [Pixhawk 6C with PM02 V3](https://holybro.com/collections/autopilot-flight-controllers/products/pixhawk-6c?variant=42420293370045)

> - [Pixhawk 6C mini with PM02 V3](https://holybro.com/collections/autopilot-flight-controllers/products/pixhawk-6c-mini?variant=42783211290813)

Reviewer: FYI I'm going to modify these in a post-process to link to internal docs for the various components. They break less often than those for external vendors.

## Additional accessories

- [GPS F9P (included in Skynode eval. kit)](https://holybro.com/products/h-rtk-f9p-gnss-series?variant=41466787102909)
- [GPS M9N (cheaper alternative to F9P)](https://holybro.com/collections/standard-gps-module/products/m9n-gps)

> - [Airspeed sensor (included in Skynode eval. kit)](https://www.dualrc.com/parts/airspeed-sensor-sdp33)

Reviewer: ```suggestion
- [Airspeed sensor (included in Skynode eval. kit)](https://www.dualrc.com/parts/airspeed-sensor-sdp33) — recommended for improved safety and performance
```

- [Airspeed sensor (cheaper alternative)](https://holybro.com/products/digital-air-speed-sensor?pr_prod_strat=use_description&pr_rec_id=236dfda00&pr_rec_pid=7150470561981&pr_ref_pid=7150472462525&pr_seq=uniform)

> - [Lidar Lightware lw20 (included in Skynode eval. kit)](https://lightwarelidar.com/products/lw20-c-100-m)

Reviewer: ```suggestion
- [Lidar Lightware lw20 (included in Skynode eval. kit)](https://lightwarelidar.com/products/lw20-c-100-m) (Optional)
```

- [Lidar (cheaper alternative)](https://www.seeedstudio.com/PSK-CM8JL65-CC5-Infrared-Distance-Measuring-Sensor-p-4028.html)
- [5V BEC](http://www.mateksys.com/?portfolio=bec12s-pro)
- RC receiver of your preference
- [Servo cable extension cable male 30cm 10 pcs](https://www.getfpv.com/male-to-male-servo-extension-cable-twisted-22awg-jr-style-5-pcs.html)
- [USB-C extension cable](https://www.digitec.ch/en/s1/product/powerguard-usb-c-usb-c-025-m-usb-cables-22529949?dbq=1&gclid=Cj0KCQjw2cWgBhDYARIsALggUhrh-z-7DSU0wKfLBVa8filkXLQaxUpi7pC0ffQyRzLng8Ph01h2R1gaAp0mEALw_wcB&gclsrc=aw.ds)
- [3M VHB tape](https://www.amazon.in/3M-VHB-Tape-4910-Length/dp/B00GTABM3Y)
- [3D-Printed mounts]()
    - 2x wing connector mount
    - 1x Airspeed sensor mount
    - 1x GPS-Mount
    - 1x Lidar-Mount
    - 1x Skynode mount
- [USB camera (included in Skynode dev kit)](https://www.amazon.com/ELP-megapixel-surveillance-machine-monitor/dp/B015FIKTZC)
- Screws, inserts, heat shrink,  etc.

## Tools

The following tools were used for this build.

- Hex driver set
- Wrench set
- Soldering station
- Glue: Hot glue, 5 min Epoxy
- Tape
- Sandpaper
- 3D-Printer

> ## Hardware integration

Reviewer: ```suggestion
## Hardware Integration

```

ZMO FPV in it's original state.

![ZMO 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/zmo-01.jpg)

Flight controller and wing connectors removed from the vehicle.

![ZMO 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/zmo-02.jpg)

### ESCs

Unsolder the PWM-signal and ground pins of the ESC and solder some servo extension wire to the pins. The cable should be long enough to connect the wire to the FMU pins of the flight controller. Unsolder the 3 female banana plug connectors of the rear motor (might not be necessary for the Pixhawk 6 integration). Screw the ESC back in place with 4 M2.5 x 12 screws. Shorten the rear motor wires and solder them as shown in the picture into place.

Solder signal and GND wires to the PWM input ot the ESC.

![ESC 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/esc-01.jpg)

Remove the female banana plug of the ESC. This will give you more space to install the flight controller.

![ESC 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/esc-02.jpg)

Solder the rear motor wires to the ESC. Make sure to have to order correct, so that the motor spins in the correct direction.

![ESC 03](../../assets/airframes/vtol/omp_hobby_zmo_fpv/esc-03.jpg)

> ### Wing connector

Reviewer: ```suggestion
### Wing Connector
```

To directly connect the wing connector when the wing gets attached, some 3D-printed mounts are needed to center the connector. This step is not necessary but makes the handling much easier and there is one step less you need to worry about when you mount the plane in the field. Glue the wing connectors into the 3D-Printed part with hot-glue or 5 min epoxy. Glue the 3D-printed part with the connector in to the fuselage. Make sure to properly align the connector while the glue cures. The easiest way to align the connector is to mount the wing while the glue is curing, but make sure that no glue is between the fuselage and the wing, otherwise the wing might get stuck.

The connector glued into the 3D-Printed part

![Wing connector 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/wing-connector-01.jpg)

The connector glued into the fuselage. Make sure to properly align the connector.

![Wing connector 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/wing-connector-02.jpg)

> ### Pixhawk adapter boards and BEC

Reviewer: ```suggestion
### Pixhawk Adapter Boards and BEC
```

![Foam cutout 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/foam-cut-01.jpg)
![Pixhawk adapter board mount 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/pixhawk-adapter-01.jpg)

Prepare the BEC to connect to the IO board and to the battery. The BEC can also be soldered directly to the battery pads of the ESC.

![BEC 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/bec-01.jpg)

Mount the BEC with double sided tape.

![BEC 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/bec-02.jpg)

### Cables

Cut off the connectors of the servos and solder the servo extension cables to the cables. Make sure that the cables are long enough to reach the Pixhawk adapter board. If you own a crimp tool, then you can also directly add the connectors without soldering. Plug the servo cables into the adapter IO board in the following order:
- 1 - Aileron left
- 2 - Aileron right
- 3 - V-Tail left
- 4 - V-Tail right
- 5 - Tilt left
- 6 - Tilt right

Pug in the motor signal cables into the FMU adapter board in the following order:
- 1 - front left
- 2 - front right
- 3 - rear

### Sensors

> #### Pitot tube

Reviewer: ```suggestion
#### Pitot Tube
```

Check first if the pitot tube fits into the 3D-Printed mount. If this is the case, glue the pitot tube mount into place. To align the tube feed it through the second hole from the right of the FPV front plate. The mount will enable you to push the tube back into the fuselage to protect it during transportation and handling. The sensor unit can be mounted on top of the 3D-Printed mount with double sided tape.

Glue the 3D-Printed mount into place.

![Pitot tube 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/pitot-tube-01.jpg)

The sensor can be mounted on top of the 3D-Printed mount.

![Pitot tube 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/pitot-tube-02.jpg)

#### Lidar

If needed a lidar can be installed in the front of the fuselage. To install the Lidar, remove the heat sink, and glue the lidar together with the 3D-Printed lidar mount into place.

![Lidar 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lidar-01.jpg)

> #### GPS

Reviewer: Yes, given that this is what is used

```suggestion
#### GPS/Compass
```

To mount the GPS, screw the 2 3D-Printed parts with 3x M3x10 screws together. Take the GPS out of the plastic case and unplug the connector. Feed the cable through the carbon spar. Glue the 3D-Printed part with 5 min epoxy in place. After the glue cured, screw the GPS with 4x M2.5x10 screws to the plate.

Glue the GPS mount into place

![GPS 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/gps-01.jpg)

Screw the GPS to the mount

![GPS 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/gps-02.jpg)

#### USB-Camera

Cut the USB cable of the camera so that the length is 15 cm. Cut the USB-Adapter cable to be 25 cm and solder the two cables together. To install the camera you need to cut a hole into the foam of the wall. Then you can mount the camera with double sided tape to the wall.

Hole to feed the USB cable through the wall.

![USB Camera 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/camera-01.jpg)

> ### Flight controller

Reviewer: ```suggestion
### Flight Controller
```

If a Skynode is used, place it at the on top of the ESCs and mark the 2 rear mounting locations on the injection molded plastic part of the ZMO. Remove the Skynode from the vehicle and drill 2 holes with a 2.8 mm drill bit into the plastic part. Put the Skynode back into place and screw it down with 2x M3x10 screws. An other option is to add some threaded inserts into the holes. Since the injection molded part of the ZMO is very thin, they need to be glued in place. Screw the front Skynode mount with 2x M3x10 screws at the Skynode. Then add some 5 min epoxy at the bottom of the mount and put a weight on top of the Skynode until the glue is cured. To better reach the 2 mounting screws at the front, poke 2 holes from the top through the foam so you can access the screws.

Mounting holes for the Skynode in the back

![Flight controller 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/flight-controller-01.jpg)

Skynode mount in the front

![Flight controller 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/flight-controller-02.jpg)

> ### Antennas and RC receiver

Reviewer: ```suggestion
### Antennas and RC Receiver

```

> One LTE antenna can be installed at the bottom of the vehicle. For that you can feed the antenna wire through the opening for the ESC heat-sink. The second antenna can be installed on the inside of the vehicle on the left side of the battery compartment. The RC receiver can also be placed at the left side of the battery compartment.

Reviewer 2: Specify that the LTE antennas are from the Skynode LTE module, and that builds with a Pixhawk instead would require some other telemetry link.

LTE antenna 1

![LTE antenna 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-01.jpg)

LTE antenna 2 and RC receiver

![LTE antenna 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-02.jpg)

## Software setup

### Select airframe

Open QGC and go under vehicle setup. Search for SYS_AUTO in the parameters tab. Set SYS_AUTOCONFIG to "Reset parameters to airframe defaults" and SYS_AUTOSTART to 13300. Reboot the vehicle.

> 

Reviewer 2: As we won't have the specific airframe selectable we should link a file with the params here.

Reviewer 2: Let's stream line this: My proposal is the following workflow:
1) open QGC and "Load from file" --> load base param file that we provide, which already includes the SYS_AUTOSTART, actuator mapping, tuning etc.
2) change actuator mapping if the user has a different wiring
3) trimming
4) sensor calibration, RC calibration
5) bench test, flying etc

Reviewer: @sfuhrer This works for me. The only thing is that I'd still like to know WHY you can't set the airframe from QGC - but are setting SYS_AUTOSTART. See https://github.com/PX4/PX4-user_guide/pull/2772#discussion_r1346491154

I'll merge once you guys have made this tidy.

Reviewer 2: That has been adapted by Tizian now. It's just a slightly different workflow

![SYS_AUTO](../../assets/airframes/vtol/omp_hobby_zmo_fpv/sys_auto.jpg)

> ### Sensor calibration

Reviewer: ```suggestion
### Sensor Calibration

```

Do the sensor calibration. Make sure to select the correct orientation of the flight controller.

### RC-Setup

Calibrate you RC controller and setup the flight mode switch.

> ### Actuator setup

Reviewer: ```suggestion
### Actuator Setup
```

:::note
Make sure the props are removed! The motors can start very easily in the actuators tab if a value gets changed in the wrong location.
:::

> #### Tilt servos

Reviewer: ```suggestion
#### Tilt Servos
```

Switch the vehicle into manual mode. Either via the flight mode switch or type "commander mode manual" into the mavlink shell. Check if the motors are pointing upwards. If they are pointing forwards they need to be reversed. To do so select the checkbox under PWM-Main for each servo. Adjust the minimum or maximum value that the servo is pointing vertical up. Then type "commander transition" into the mavlink shell to adjust the horizontal position.

![Tilt Servo adjustment](../../assets/airframes/vtol/omp_hobby_zmo_fpv/tilt-limits-01.jpg)

> #### Control surfaces

Reviewer: ```suggestion
#### Control Surfaces
```

Check with RC-Controller, if the actuators need to be reversed.
- Roll stick to the right -> Right aileron up, left aileron down
- Pitch stick to the back (fly upwards) -> both V-tail surfaces up
- Yaw stick to the right -> both surfaces to the right

Now adjust the trim value that all the surfaces are in neutral position.

![Servo trim](../../assets/airframes/vtol/omp_hobby_zmo_fpv/servo-trim.jpg)

> #### Motor direction and orientation

Reviewer: ```suggestion
#### Motor Direction and Orientation
```

Make sure the props are removed!!!

- Motor 1: Front left motor should spin CW
- Motor 2: Front right motor should spin CCW
- Motor 3: Rear motor should spin CCW

> If the motor spins in the wrong directions 2 of the 3 motor wires need to be swapped. The direction can not be changed in software since the ESCs are not using dshot.

Reviewer 2: ```suggestion
If the motor spins in the wrong directions two of the three motor wires need to be swapped. The direction can not be changed in software since the ESCs are not using dshot.

```

Reviewer: FWIW What @sfuhrer has done is correct English prose. I do not enforce this particular case in PX4 docs - we use "pragmatic English prose". 

Why? Because there is no ambiguity in the use of numbers here, and it reduces effort to translate. However I do accept fixes like this if particular reviewers want them.

> ## First flight

Reviewer: ```suggestion
## First Flight
```

- Check CG (There are markings underneath the wing)
- Check actuator orientations and neutral trim

> - Check tilt rotor reactions in stabilized

Reviewer 2: ```suggestion
- Check tilt rotor reactions in Stabilized flight mode

```

    - Yaw the vehicle to the right (nose to the right) -> right motor should tilt down
    - Yaw the vehicle to the left (nose to the left) -> left motor should tilt down

> - Check actuator reactions in stabilized (switch transition switch to enable fw mode)

Reviewer 2: ```suggestion
- Check actuator reactions in Stabilized (switch transition switch to enable FW mode)

```

Reviewer: FYI, note that where possible, good to link to the mode you talk about - that's even better than using the proper name. I have modified the comment above and will merge it.

    - Roll the vehicle to the right -> Right aileron should go down
    - Pitch the vehicle up (nose up) -> both elevons should go down
    - Yaw the vehicle to the right (nose to the right) -> both elevons should go to the right, 
- Check kill switch

> - Arm in stabilized and check if motors react in the correct orientation but do not take off

Reviewer 2: ```suggestion
- Arm in stabilized and check if motors respond to the commands, e.g. roll left increases throttle on the right motor

```

- Takeoff in stabilized and make some basic maneuvers

> - If everything went without any issue, takeoff in position mode and do a transition at ca. 25m. If something goes wrong switch back to MC mode as fast as possible.

Reviewer 2: ```suggestion
- If everything went without any issue, takeoff in Position mode and do a transition at ca. 50m. If something goes wrong switch back to MC mode as fast as possible.
```

If a Pixhawk 6c or 6c mini is used, simply stick the flight controller with double sided tape into place.

If a Skynode is used, place it at the on top of the ESCs and mark the 2 rear mounting locations on the injection molded plastic part of the ZMO.
Remove the Skynode from the vehicle and drill 2 holes with a 2.8 mm drill bit into the plastic part.
Put the Skynode back into place and screw it down with 2x M3x10 screws.
An other option is to add some threaded inserts into the holes. Since the injection molded part of the ZMO is very thin, they need to be glued in place. Screw the front Skynode mount with 2x M3x10 screws at the Skynode.
Then add some 5 min epoxy at the bottom of the mount and put a weight on top of the Skynode until the glue is cured.
To better reach the 2 mounting screws at the front, poke 2 holes from the top through the foam so you can access the screws.

Mounting holes for the Skynode in the back

![Flight controller 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/flight-controller-01.jpg)

Skynode mount in the front

![Flight controller 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/flight-controller-02.jpg)

> ### Antennas and RC Receiver

Reviewer: Yes, this section needs info about the options here - LTE is only for Skynode right?

Author: ```suggestion
### Antennas and RC Receiver

:::note
If a Skynode is installed the LTE can be used as telemetry and video link. If a Pixhawk is used a different telemetry link will be needed. A basic example would be a [SiK Telemetry Radio](https://holybro.com/products/sik-telemetry-radio-v3?_pos=2&_sid=e5e1c6ba3&_ss=r).
:::

```
I added this note that a seperate telemetry link is needed if a Pixhawk is used.

Reviewer: Thanks. I'll modify a little - where possible better to link to internal docs - such as for Sik radio.

One LTE antenna can be installed at the bottom of the vehicle. For that you can feed the antenna wire through the opening for the ESC heat-sink.
The second antenna can be installed on the inside of the vehicle on the left side of the battery compartment.
The RC receiver can also be placed at the left side of the battery compartment.

LTE antenna 1

![LTE antenna 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-01.jpg)

LTE antenna 2 and RC receiver

![LTE antenna 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-02.jpg)

## Software Setup

### Select Airframe

> 1. Open QGC and go under vehicle setup.

Reviewer: @sfuhrer Why are we doing all this? Why not jus select a generic vtol tiltrotor in the Airframe selection screen, then configure the geometry? (i.e. remove one motor)

https://docs.px4.io/main/en/airframes/airframe_reference.html#vtol_vtol_tiltrotor_generic_quadplane_vtol_tiltrotor

I do think it would be good to provide a parameter file that can be applied as a diff if people want

1. Search for [SYS_AUTO](../advanced_config/parameter_reference.md#SYS_AUTOSTART) in the parameters tab.
1. Set [SYS_AUTOCONFIG](../advanced_config/parameter_reference.md#SYS_AUTOCONFIG) to "Reset parameters to airframe defaults" and [SYS_AUTOSTART](../advanced_config/parameter_reference.md#SYS_AUTOSTART) to `13300`.
1. Reboot the vehicle.

![SYS_AUTO](../../assets/airframes/vtol/omp_hobby_zmo_fpv/sys_auto.png)

### Sensor Calibration

First make sure to set the [correct orientation of the flight controller](../config/flight_controller_orientation.md).
This should be the default (`ROTATION_NONE`).

> Then calibrate the main sensors:

Reviewer: FYI I modified this a bit to explain in more detail what you do - more importantly to point to the relevant docs. NOTE, I am assuming mounting is as ROTATION_NONE

- Motor 2: Front right motor should spin CCW
- Motor 3: Rear motor should spin CCW

If the motor spins in the wrong directions two of the three motor wires need to be swapped. The direction can not be changed in software since the ESCs are not using dshot.

## First Flight

- Check CG (There are markings underneath the wing)
- Check actuator orientations and neutral trim
- Check tilt rotor reactions in [Stabilized mode](../flight_modes_fw/stabilized.md)
  - Yaw the vehicle to the right (nose to the right) -> right motor should tilt down
  - Yaw the vehicle to the left (nose to the left) -> left motor should tilt down
- Check tilt rotor reactions in [Stabilized mode](../flight_modes_fw/stabilized.md)
  - Roll the vehicle to the right -> Right aileron should go down
  - Pitch the vehicle up (nose up) -> both elevons should go down
  - Yaw the vehicle to the right (nose to the right) -> both elevons should go to the right,
- Check kill switch
- Arm in [Stabilized mode](../flight_modes_fw/stabilized.md) and check if motors respond to the commands, e.g. roll left increases throttle on the right motor
- Takeoff in stabilized and make some basic maneuvers

> - If everything went without any issue, takeoff in [Position mode](../flight_modes_fw/position.md) and do a transition at around 50m. If something goes wrong switch back to MC mode as fast as possible.

Reviewer: What about tuning? SHould we autotune?

Reviewer 2: No we'll provide recommended PX4 params including the tuning.

Reviewer: Sounds good, but what's the workflow? Are we expecting them to load airframe, then params then tidy up the actuator output mapping for how they wired it, or is this just something we provide at the end?

NOrmally we'd say "do all this, including tuning" then "here is a param file we provided earlier". Why? Because I'd like to make sure our tuning information is followable for this.

Reviewer 2: You're right, it's a bit contradicting atm. I've proposed a workflow that should work in https://github.com/PX4/PX4-user_guide/pull/2772/files#r1346984309

If a Pixhawk 6c or 6c mini is used, simply stick the flight controller with double sided tape into place.

If a Skynode is used, place it at the on top of the ESCs and mark the 2 rear mounting locations on the injection molded plastic part of the ZMO.
Remove the Skynode from the vehicle and drill 2 holes with a 2.8 mm drill bit into the plastic part.
Put the Skynode back into place and screw it down with 2x M3x10 screws.
An other option is to add some threaded inserts into the holes. Since the injection molded part of the ZMO is very thin, they need to be glued in place. Screw the front Skynode mount with 2x M3x10 screws at the Skynode.
Then add some 5 min epoxy at the bottom of the mount and put a weight on top of the Skynode until the glue is cured.
To better reach the 2 mounting screws at the front, poke 2 holes from the top through the foam so you can access the screws.

Mounting holes for the Skynode in the back

![Flight controller 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/flight-controller-01.jpg)

Skynode mount in the front

![Flight controller 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/flight-controller-02.jpg)

### Antennas and RC Receiver

:::note

> If a Skynode is installed the LTE can be used as telemetry and video link. If a Pixhawk is used a different telemetry link will be needed. A basic example would be a [SiK Telemetry Radio](https://holybro.com/products/sik-telemetry-radio-v3?_pos=2&_sid=e5e1c6ba3&_ss=r).

Reviewer: ```suggestion
If a Skynode is installed the LTE can be used as telemetry and video link.
If a Pixhawk is used a different [telemetry link](../telemetry/README.md) will be needed.
An inexpensive example would be a [SiK Telemetry Radio](../telemetry/sik_radio.md).
```

![LTE antenna 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-01.jpg)

LTE antenna 2 and RC receiver

![LTE antenna 02](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-02.jpg)

## Software Setup

### Select Airframe

1. Open QGC and go under vehicle setup.

1. Open QGC and go under vehicle setup.
1. Search for [SYS_AUTO](../advanced_config/parameter_reference.md#SYS_AUTOSTART) in the parameters tab.
1. Set [SYS_AUTOCONFIG](../advanced_config/parameter_reference.md#SYS_AUTOCONFIG) to "Reset parameters to airframe defaults" and [SYS_AUTOSTART](../advanced_config/parameter_reference.md#SYS_AUTOSTART) to `1230113`.
1. Reboot the vehicle.
1. Load the [parameter-file]() up to the vehicle. To do so, go into the parameter tab and press on tools in the top-right corner. Select "Load from file.." and select the file.

### Sensor Calibration

First make sure to set the [correct orientation of the flight controller](../config/flight_controller_orientation.md).
This should be the default (`ROTATION_NONE`).

Then calibrate the main sensors:

- [Compass](../config/compass.md)
- [Gyroscope](../config/gyroscope.md)
- [Accelerometer](../config/accelerometer.md)

> - [Airspeed](../config/airspeed.html)

Reviewer: ```suggestion
- [Airspeed](../config/airspeed.md)

```

### Antennas and RC Receiver

:::note
If a Skynode is installed the LTE can be used as telemetry and video link.
If a Pixhawk is used a different [telemetry link](../telemetry/README.md) will be needed.
An inexpensive example would be a [SiK Telemetry Radio](../telemetry/sik_radio.md).
:::

1. One LTE antenna can be installed on the bottom of the vehicle.
   For that you can feed the antenna wire through the opening for the ESC heat-sink.

   ![LTE antenna 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-01.jpg)

1. The second antenna can be installed on the inside of the vehicle on the left side of the battery compartment.
   The RC receiver can also be placed at the left side of the battery compartment.

   ![LTE antenna 2 and RC receiver](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-02.jpg)

## Software Setup

### Select Airframe

1) Open QGC and go under vehicle setup.
1) Select [Generic Quadplane VTOL](https://docs.px4.io/main/en/airframes/airframe_reference.html#vtol_standard_vtol_generic_standard_vtol) under the [Airframe](https://docs.px4.io/main/en/config/airframe.html) tab and click "Apply and Restart"

> 1) Select the [Parameters](https://docs.px4.io/main/en/advanced_config/parameters.html#finding-updating-parameters) tab and click on "Tools" in the top right corner. Select "Load from file..". Download the [parameter file](https://github.com/PX4/PX4-user_guide/raw/main/assets/airframes/vtol/omp_hobby_zmo_fpv/OMP-Hobby-ZMO.params) and upload it to the flight controller.
> 1) reboot the vehicle

Reviewer: Markdown expects lists to look like this `1.`  (though you can order them, you don't have to)
```suggestion
1. Select the [Parameters](https://docs.px4.io/main/en/advanced_config/parameters.html#finding-updating-parameters) tab and click on "Tools" in the top right corner. Select "Load from file..". Download the [parameter file](https://github.com/PX4/PX4-user_guide/raw/main/assets/airframes/vtol/omp_hobby_zmo_fpv/OMP-Hobby-ZMO.params) and upload it to the flight controller.
1. Reboot the vehicle.
```

- [Airspeed sensor (cheaper alternative)](https://holybro.com/products/digital-air-speed-sensor?pr_prod_strat=use_description&pr_rec_id=236dfda00&pr_rec_pid=7150470561981&pr_ref_pid=7150472462525&pr_seq=uniform)
- [Lidar Lightware lw20-c (included in Skynode eval. kit)](../sensor/sfxx_lidar.md) (Optional)
- [Lidar Seeed Studio PSK-CM8JL65-CC5 (cheaper alternative)](https://www.seeedstudio.com/PSK-CM8JL65-CC5-Infrared-Distance-Measuring-Sensor-p-4028.html)
- [5V BEC](http://www.mateksys.com/?portfolio=bec12s-pro)
- [Radio (RC) System](../getting_started/rc_transmitter_receiver.md) of your preference
- [Servo cable extension cable male 30cm 10 pcs](https://www.getfpv.com/male-to-male-servo-extension-cable-twisted-22awg-jr-style-5-pcs.html)
- [USB-C extension cable](https://www.digitec.ch/en/s1/product/powerguard-usb-c-usb-c-025-m-usb-cables-22529949?dbq=1&gclid=Cj0KCQjw2cWgBhDYARIsALggUhrh-z-7DSU0wKfLBVa8filkXLQaxUpi7pC0ffQyRzLng8Ph01h2R1gaAp0mEALw_wcB&gclsrc=aw.ds)
- [3M VHB tape](https://www.amazon.in/3M-VHB-Tape-4910-Length/dp/B00GTABM3Y)

> - [3D-Printed mounts](https://github.com/PX4/PX4-user_guide/raw/main/assets/airframes/vtol/omp_hobby_zmo_fpv/OMP-Hobby-ZMO-3D-Prints.zip)

Reviewer: FYI only -note this changed to a raw link rather than linking to the asset locally as you had done. Vuepress can't handle a direct link for some reason.

  - 2x wing connector mount
  - 1x Airspeed sensor mount
  - 1x GPS-Mount
  - 1x Lidar-Mount
  - 1x Skynode mount
- [USB camera (included in Skynode dev kit)](https://www.amazon.com/ELP-megapixel-surveillance-machine-monitor/dp/B015FIKTZC)
- Screws, inserts, heat shrink, etc.

## Tools

The following tools were used for this build.

- Hex driver set
- Wrench set
- Soldering station
- Glue: Hot glue, 5 min Epoxy
- Tape
- 3M Double sided tape ([3M VHB tape](https://www.amazon.in/3M-VHB-Tape-4910-Length/dp/B00GTABM3Y))
- Sandpaper
- 3D-Printer

## Hardware Integration

### Preparations

Remove the original flight controller, ESC and wing connector cables.
Also remove the the propellers.
This will help you with the handling of the vehicle and will reduce the risk of an injury due to an unintentional motor startup.

ZMO FPV in it's original state.

![ZMO FPV in it's original state](../../assets/airframes/vtol/omp_hobby_zmo_fpv/zmo-01.jpg)

Flight controller and wing connectors removed from the vehicle.

![ZMO FPV with FC and wing connectors removed](../../assets/airframes/vtol/omp_hobby_zmo_fpv/zmo-02.jpg)

### ESCs

> 1. Unsolder the ESC PWM-signal and ground pins and solder some servo extension wire to the pins.

Reviewer: FYI @tizian07 All the way through you will note I have modified this to a list. I prefer this because otherwise the instructions can be a "wall of text". By putting in list you can check that the instructions make sense in order, and put the images associated with each step where it is relevant.

It isn't "always better" but it can make these things easier to follow.

:::note
If a Skynode is installed the LTE can be used as telemetry and video link.
If a Pixhawk is used a different [telemetry link](../telemetry/README.md) will be needed.
An inexpensive example would be a [SiK Telemetry Radio](../telemetry/sik_radio.md).
:::

1. One LTE antenna can be installed on the bottom of the vehicle.
   For that you can feed the antenna wire through the opening for the ESC heat-sink.

   ![LTE antenna 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-01.jpg)

1. The second antenna can be installed on the inside of the vehicle on the left side of the battery compartment.
   The RC receiver can also be placed at the left side of the battery compartment.

   ![LTE antenna 2 and RC receiver](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-02.jpg)

## Software Setup

### Select Airframe

1. Open QGC, select the **Q** icon, and then select **Vehicle Setup**.
1. Select the [Airframe](https://docs.px4.io/main/en/config/airframe.html) tab
1. Select [Generic Tiltrotor VTOL](../airframes/airframe_reference.md#vtol_vtol_tiltrotor_generic_tiltrotor_vtol) from the _VTOL Tiltrotor_ group, and then and click **Apply and Restart**.

> ### Load Parameters File

Reviewer: FYI, added to own section - allows explanation and linking.

Note strictly I guess you don't need the airframe loading if you have this, but I think it is good to show both.

To load the file:

1. Download the [parameter file](https://github.com/PX4/PX4-user_guide/raw/main/assets/airframes/vtol/omp_hobby_zmo_fpv/OMP-Hobby-ZMO.params).
1. Select the [Parameters](../advanced_config/parameters.md#finding-updating-parameters) tab and then click on **Tools** in the top right corner.
1. Select **Load from file** and then choose the `OMP-Hobby-ZMO.params` file you just downloaded.
1. Reboot the vehicle.

### Sensor Calibration

First make sure to set the [correct orientation of the flight controller](../config/flight_controller_orientation.md).
This should be the default (`ROTATION_NONE`).

Then calibrate the main sensors:

- [Compass](../config/compass.md)
- [Gyroscope](../config/gyroscope.md)
- [Accelerometer](../config/accelerometer.md)
- [Airspeed](../config/airspeed.md)

### RC-Setup

[Calibrate your RC Controller](../config/radio.md) and setup the [flight mode switches](../config/flight_mode.md).

We recommend you assign RC switches for the set of modes defined in [Flight Mode Configuration > What Flight Modes and Switches Should I Set?](../config/flight_mode.md#what-flight-modes-and-switches-should-i-set).

> In particular you should assign a _VTOL Transition Switch_, _Kill Switch_, and a switch to select [Stabilized mode](../flight_modes_fw/stabilized.md) and [Position mode](../flight_modes_fw/position.md).

Reviewer: FYI added this because you later require certain switches to be set.

### Actuator Setup

:::warning
Make sure the props are removed!
The motors are easy to start in the actuators tab by accident.
:::

Motors, control surfaces, and other actuators are configured in the QGroundControl [Actuator Configuration & Testing](../config/actuators.md).

The [parameter file](#load-parameters-file) loaded previously means that this screen should already be correctly setup: you just need to adjust the trims for your particular vehicle.

> If motors/servos were connected to different outputs than suggested, you will need to change the output to function mappings in the actuator output section.

Reviewer: FYI this explains (again) why we're not doing geometry etc, and points out you might need to change wiring.

   ![Tilt Servo adjustment](../../assets/airframes/vtol/omp_hobby_zmo_fpv/tilt-limits-01.jpg)

1. Adjust the minimum or maximum value that the servo is pointing vertical up.
1. Then type `commander transition` into the MAVLink shell to adjust the horizontal position.

#### Control Surfaces

Check if the actuators need to be reversed using the RC-Controller:

- Roll stick to the right. The right aileron should go up, left aileron should go down.
- Pitch stick to the back (fly upwards). Both V-tail surfaces should move up.
- Yaw stick to the right. Both surfaces should move to the right

Now adjust the trim value so that all the surfaces are in neutral position.

![Servo trim](../../assets/airframes/vtol/omp_hobby_zmo_fpv/servo_trim.png)

#### Motor Direction and Orientation

Make sure the props are removed!!

- `Motor 1`: Front left motor should spin CW
- `Motor 2`: Front right motor should spin CCW
- `Motor 3`: Rear motor should spin CCW

If the motor spins in the wrong directions two of the three motor wires need to be swapped.
The direction can't be changed in software because the vehicle does not use [DShot ESC](../peripherals/dshot.md).

## First Flight

> - Check CG (There are markings underneath the wing)

Reviewer: Center of Gravity? Exactly what is it that you expect me to check?

- Check actuator orientations and neutral trim
- Check tilt rotor reactions in [Stabilized mode](../flight_modes_fw/stabilized.md)
  - Yaw the vehicle to the right (nose to the right). The right motor should tilt down.
  - Yaw the vehicle to the left (nose to the left). The left motor should tilt down.
- Check tilt rotor reactions in [Stabilized mode](../flight_modes_fw/stabilized.md)
  - Roll the vehicle to the right. The right aileron should go down.
  - Pitch the vehicle up (nose up). Both elevons should go down.
  - Yaw the vehicle to the right (nose to the right). Both elevons should go to the right.

> - Check kill switch

Reviewer: Why? How? A beginner might flight to 300m then press this. Do you think we could provide some more advice?

- Arm in [Stabilized mode](../flight_modes_fw/stabilized.md) and check if motors respond to the commands, e.g. roll left increases throttle on the right motor
- Takeoff in [Stabilized mode](../flight_modes_fw/stabilized.md) and make some basic maneuvers
- If everything went without any issue, takeoff in [Position mode](../flight_modes_fw/position.md) and do a transition at around 50m.

>   If something goes wrong switch back to multicopter mode as fast as possible.

Reviewer: ```suggestion
  If something goes wrong switch back to multicopter mode as fast as possible (using the transition switch).

```

   To access the 2 mounting screws at the front, poke 2 holes from the top through the foam.

   ![Skynode mount in the front](../../assets/airframes/vtol/omp_hobby_zmo_fpv/flight-controller-02.jpg)

### Antennas and RC Receiver

:::note
If a Skynode is installed the LTE can be used as telemetry and video link.
If a Pixhawk is used a different [telemetry link](../telemetry/README.md) will be needed.
An inexpensive example would be a [SiK Telemetry Radio](../telemetry/sik_radio.md).
:::

1. One LTE antenna can be installed on the bottom of the vehicle.
   For that you can feed the antenna wire through the opening for the ESC heat-sink.

   ![LTE antenna 01](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-01.jpg)

1. The second antenna can be installed on the inside of the vehicle on the left side of the battery compartment.
   The RC receiver can also be placed at the left side of the battery compartment.

   ![LTE antenna 2 and RC receiver](../../assets/airframes/vtol/omp_hobby_zmo_fpv/lte-antenna-02.jpg)

## Software Setup

### Select Airframe

1. Open QGC, select the **Q** icon, and then select **Vehicle Setup**.

> 1. Select the [Airframe](https://docs.px4.io/main/en/config/airframe.html) tab

Reviewer: ```suggestion
1. Select the [Airframe](../config/airframe.md) tab

```

### Select Airframe

1. Open QGC and go under vehicle setup.

1. Open QGC and go under vehicle setup.
1. Search for [SYS_AUTO](../advanced_config/parameter_reference.md#SYS_AUTOSTART) in the parameters tab.

> 1. Set [SYS_AUTOCONFIG](../advanced_config/parameter_reference.md#SYS_AUTOCONFIG) to "Reset parameters to airframe defaults" and [SYS_AUTOSTART](../advanced_config/parameter_reference.md#SYS_AUTOSTART) to `1230113`.

Reviewer 2: 13100 would be the generic tiltrortor VTOL, no?

Reviewer: My point in https://github.com/PX4/PX4-user_guide/pull/2772/files#r1345065617 is that you shouldn't do any of this. YOu just select the generic airframe by name in the config screen https://docs.px4.io/main/en/config/airframe.html then adjust the geometry.

### Select Airframe

1) Open QGC and go under vehicle setup.

> 1) Select [Generic Quadplane VTOL](https://docs.px4.io/main/en/airframes/airframe_reference.html#vtol_standard_vtol_generic_standard_vtol) under the [Airframe](https://docs.px4.io/main/en/config/airframe.html) tab and click "Apply and Restart"

Reviewer 2: Should be "Generic Tiltrotor VTOL"

Reviewer: I think so.
```suggestion
1. Select [Generic Tiltrotor VTOL](../airframes/airframe_reference.md#vtol_vtol_tiltrotor_generic_tiltrotor_vtol) under the [Airframe](https://docs.px4.io/main/en/config/airframe.html) tab and click **Apply and Restart**.
```
